src/main/java/core/basesyntax/model/FruitTypeFactory.java

```
@@ -0,0 +1,7 @@
+package core.basesyntax.model;
+
+public interface FruitTypeFactory {
+    default FruitType create(String name) {
+        return new FruitType(name);
+    }
```

Providing the factory as an interface default method is acceptable, but embedding creation logic without validation can hide errors. Consider moving the creation logic (with validation) into a concrete implementation (e.g., `DefaultFruitTypeFactory`) and have callers depend on the `FruitTypeFactory` abstraction. This better follows the requirement: "High-level modules should depend on abstractions (interfaces), not concrete implementations." Implementations can then throw informative RuntimeExceptions for invalid input instead of allowing NPEs later.